cmake/Dependencies.cmake

```
@@ -1024,11 +1024,18 @@ if(BUILD_PYTHON)
   endif()
 
   if(PYTHONINTERP_FOUND AND PYTHONLIBS_FOUND)
-    include_directories(SYSTEM ${PYTHON_INCLUDE_DIR})
+    add_library(python::python INTERFACE IMPORTED)
+    set_property(TARGET python::python PROPERTY
+        INTERFACE_INCLUDE_DIRECTORIES ${PYTHON_INCLUDE_DIRS})
+    set_property(TARGET python::python PROPERTY
+        INTERFACE_LINK_LIBRARIES ${PYTHON_LIBRARIES})
```

I see that the link `PYTHON_LIBRARIES` is guarded  with `if(WIN32)` elseswhere. Maybe this would fix it:
```suggestion
    if(WIN32)
      set_property(TARGET python::python PROPERTY
          INTERFACE_LINK_LIBRARIES ${PYTHON_LIBRARIES})
    endif()
```